backend/internal/common/env_config.go

```diff
@@ -133,6 +136,18 @@ func parseEnvConfig() error {
 		return errors.New("APP_URL must not contain a path")
 	}
 
+	// only check if internal app url is not default value
+	if EnvConfig.InternalAppURL != EnvConfig.AppURL {
```

Comment:
By making this change, you are requiring users to always set `INTERNAL_APP_URL` if they are setting `APP_URL`.

Instead:

1. Leave the default value of `InternalAppURL` (L94) empty
2. Here:
    - if the value of `InternalAppURL` is empty, set it to the value of `AppURL`
    - if it's not empty, do the validation below